src/main/java/ch/njol/skript/conditions/CondIsWhitelisted.java

```diff
@@ -19,72 +19,69 @@
 package ch.njol.skript.conditions;
 
 import org.bukkit.Bukkit;
-import org.bukkit.entity.Player;
+import org.bukkit.OfflinePlayer;
 import org.bukkit.event.Event;
 import org.eclipse.jdt.annotation.Nullable;
 
 import ch.njol.skript.Skript;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.lang.Condition;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.util.Kleenean;
 
 @Name("Is Whitelisted")
-@Description("Whether or not the server or a player is whitelisted.")
-@Examples({"if server is whitelisted:", "if player is whitelisted"})
-@Since("2.5.2")
+@Description("Whether or not the server or a player is whitelisted, or the server is whitelist enforced.")
+@Examples({
+	"if player is whitelisted:",
+	"if server is whitelisted:",
+	"if server is whitelist enforced:"
+})
+@Since("2.5.2, INSERT VERSION (enforce, offline players)")
+@RequiredPlugins("Minecraft 1.17+ (enforce)")
 public class CondIsWhitelisted extends Condition {
-	
+
+	private static final boolean ENFORCE_SUPPORT = Skript.methodExists(Bukkit.class, "isWhitelistEnforced");
+
 	static {
 		Skript.registerCondition(CondIsWhitelisted.class,
-			"[the] server (is|1¦is(n't| not)) white[ ]listed",
-			"%players% (is|are)(|1¦(n't| not)) white[ ]listed");
+			"[the] server (is|not:(isn't|is not)) white[ ]listed",
+			"%offlineplayers% (is|are)(|not:(isn't|is not)) white[ ]listed",
```

Comment:
I might be reading this wrong but could that pattern allow something like `player is is not whitelisted`

Reply:
oh yeah I think the parentheses after `are` are accidental

This should probably support `are not` and `aren't` too

Reply:
Good catch!